### PR TITLE
Feat: Search mail bodies

### PR DESCRIPTION
**Pull request by hamza221**
Fixes https://github.com/nextcloud/mail/issues/8050

- [x] Add a field in advanced search for body only search

### CODE REVIEW

File: lib/Db/MessageMapper.php [hunk 1]
```diff
@@ -832,7 +832,7 @@ public function findIdsByQuery(Mailbox $mailbox, SearchQuery $query, ?int $limit
 
 		// createParameter
 		if ($uids !== null) {
-			$select->andWhere(
+			$select->orWhere(
```
Comment (kesselb) [inline lib/Db/MessageMapper.php:835]: Debugging leftover?

**Comment by ChristophWurst** [comment]
Code looks good!

Right now there is a new dedicated input in the advanced search. We also discussed an opt-in feature to search in bodies in the search box on top of envelopes, right? I'm fine with doing another PR.

**Comment by sergioeix** [comment]
cant blive this is not option yet, very usefull. Cant wait to test it :)

**Comment by ChristophWurst** [comment]
If I search for a body term only, the search doesn't really have an affect. I also get messages without the search term. Is it the AND vs OR combination of the IMAP and DB results?

**Comment by hamza221** [comment]
> If I search for a body term only, the search doesn't really have an affect. I also get messages without the search term. Is it the AND vs OR combination of the IMAP and DB results?

It is. So if I'm correct,  the only case where we need an orWhere is when we're searching both body and subject, all other combinations should be andWhere.

File: lib/Db/MessageMapper.php [hunk 2]
```diff
@@ -832,9 +832,15 @@ public function findIdsByQuery(Mailbox $mailbox, SearchQuery $query, ?int $limit
 
 		// createParameter
 		if ($uids !== null) {
-			$select->andWhere(
-				$qb->expr()->in('m.uid', $qb->createParameter('uids'))
-			);
+			if(!empty($query->getSubjects())) {
```
Comment (ChristophWurst) [inline lib/Db/MessageMapper.php:835]: please leave a comment for this logic. one of us will wonder about this `if` in the future :)

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Works!

**Comment by HLFH** [comment]
@ChristophWurst Hi. Could we have this released in Nextcloud Mail 3.4.5? Thank you!

**Pull request merged**